.pre-commit-config.yaml

```diff
@@ -8,15 +8,19 @@ repos:
     hooks:
       - id: trailing-whitespace
       - id: end-of-file-fixer
+        exclude: |
+          (?x)^(
+              .*\.svg
+          )$
       - id: check-yaml
       - id: check-added-large-files
       - id: detect-private-key
   - repo: https://github.com/python-poetry/poetry
-    rev: "1.5.1"
+    rev: "1.8.0"
     hooks:
       - id: poetry-check
   - repo: https://github.com/codespell-project/codespell
-    rev: v2.2.6
+    rev: v2.3.0
```

Comment:
General question, what was your process on deciding to upgrade the versions in this pre-commit file?

Reply:
Thank you for the question - my process is to consider where I'm making modifications and then try to absorb a bit of additional refactoring with those files / blocks while I'm there to "leave it better than I found it". Some of how I think can be considered ["opportunistic refactoring"](https://martinfowler.com/bliki/OpportunisticRefactoring.html) - we have to be careful about leaving [technical debt](https://en.wikipedia.org/wiki/Technical_debt) alone for too long (we don't know when we'll come back to it!).